lib/buffer.js

```diff
@@ -762,13 +767,18 @@ function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
   if (Number.isNaN(byteOffset)) {
     byteOffset = dir ? 0 : buffer.length;
   }
+  encoding = encoding || 'utf8';
   dir = !!dir;  // Cast to bool.
 
+  if (encoding in EncodingsToUseWithBufferValues && typeof val === 'string') {
```

Comment:
if you use `in` please make sure the target object is null proto

Reply:
Good point.
But I'm seriously considering moving this to CPP